CMSgov/mint-app
# [EASI-3098] Share Model Plan

# EASI-3098

## Changes and Description

- Share Model email template
- View Filter enum

FE Changes
- Added UI for `Share` on modal
- Added clipboard copy for entire model plan as well as filtered view link
- Mutation to send model plan
- Success/error states on tasklist/readonly
- Updated unit tests

<img width="692" alt="Screenshot 2023-08-29 at 10 49 59 AM" src="https://github.com/CMSgov/mint-app/assets/95709965/2cb2c943-27c3-4724-9955-d1371e12d88c">

<img width="798" alt="Screenshot 2023-08-29 at 10 50 27 AM" src="https://github.com/CMSgov/mint-app/assets/95709965/9e6535a5-a25a-4e83-a512-6f6163fadc52">

<img width="819" alt="Screenshot 2023-08-29 at 10 50 59 AM" src="https://github.com/CMSgov/mint-app/assets/95709965/33abb950-016b-4886-a294-deed382028a6">

<img width="822" alt="Screenshot 2023-08-29 at 11 03 40 AM" src="https://github.com/CMSgov/mint-app/assets/95709965/07142f0d-964e-4ded-9a5a-d19503a0851e">


## How to test this change

*From both task list and readonly

- Verify if opening modal from filtered view, that is gets autopopulated
- Verify button is disabled if not filter view is selected
- Verify mutation passes successfully if valid email - success Alert component rendered
- Verify mutation fail on inproper email address - error Alert component rendered
- Verify copy link to clipboard is successful on both all plan and filter view selections

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
